src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -153,13 +154,13 @@ private static void AsyncLocalSetCurrentUICulture(AsyncLocalValueChangedArgs<Cul
         private static CultureInfo InitializeUserDefaultCulture()
         {
             Interlocked.CompareExchange(ref s_userDefaultCulture, GetUserDefaultCulture(), null);
-            return s_userDefaultCulture;
+            return s_userDefaultCulture!;
         }
 
         private static CultureInfo InitializeUserDefaultUICulture()
         {
             Interlocked.CompareExchange(ref s_userDefaultUICulture, GetUserDefaultUICulture(), null);
-            return s_userDefaultUICulture;
+            return s_userDefaultUICulture!;
```

Comment:
This is an interesting case, @jaredpar.  In general nullability analysis explicitly ignores threading, e.g. a property that's been validated to be non-null will suppress the compiler warning:
```C#
if (obj.SomeProp != null) obj.SomeProp.SomeMethod(); // no warning
```
even though the second call to SomeProp could return null (either due to some other thread changing things, or just more simply the property not being particularly well-behaved).

This feels like a similar case, but at the same time, if you're using CompareExchange, there be dragons.  It seems like the net result here will be that lots of cases involving CompareExchange will also require using the null-forgiveness operator.

Reply:
This was a deliberate design trade off the nullable feature made. The explicit assumption was made that properties values will be non-null or `null` across various invocations. Had a healthy debate about this but eventually came down on this side.

Reply:
Right, I understand the decision about properties.  My point (which I didn't make particularly well) is that this case is similar: the line right above it is initializing the field, and then we read from the field.  Following the same logic as was used with properties, you could see an argument being made for the compiler treating it as initialized here as well, assuming GetUserDefaultUICulture is typed to return non-null.

Reply:
Ah I see.  How are you annotating `CompareExachange` first parameter here: nullable or nonnullableL

Reply:
> How are you annotating CompareExachange first parameter here

How are you allowing generics to be annotated? :-)
https://source.dot.net/#System.Private.CoreLib/src/System/Threading/Interlocked.cs,105

Regardless, you're absolutely able to pass null for either/both the second or third arguments.